canaries/2D_image_stream.py

```diff
@@ -39,44 +52,101 @@ def get_frames_per_sec(ordered_frames):
+        # This is to stop this script just before the start of the next hour.
+        current_min = int(datetime.datetime.now().strftime("%M"))
+        if current_min == 59:
+            await close_robot(robot)
+            return
+        pil_img = await cam.get_image()
+        llist1.append(datetime.datetime.now())
+
+        draw = ImageDraw.Draw(pil_img)
+        draw.rectangle((0, 0, 75, 25), outline='black')
+        fps = get_frames_per_sec(llist1)
+        draw.text(xy=(18, 7.5), text=f'{fps} FPS', fill='white')
+
+        open_cv_image = cv2.cvtColor(np.array(pil_img), cv2.COLOR_RGB2BGR)
+
+        window_name = '2D'
+        cv2.namedWindow(window_name, cv2.WND_PROP_FULLSCREEN)
+        if arg == 1:
+            print("arg = 1")
+            cv2.setWindowProperty(window_name, cv2.WND_PROP_FULLSCREEN, cv2.WINDOW_FULLSCREEN)
+        else:
+            print("arg = 3")
+            open_cv_image = open_cv_image[0:480, 0:425]
+            cv2.resizeWindow(window_name, window_width, window_height)
+            cv2.moveWindow(window_name, 0, 0)
+        cv2.imshow(window_name, open_cv_image)
+        cv2.waitKey(1)
+    return
 
 async def main():
+    print(arg)
     robot = None
-    llist = deque()
+    llist1 = deque()
+    llist2 = deque()
     try:
         robot = await connect()
-        logging.info("finished connecting to robot client")
-
-        cam_name = "standard_camera"
-        cam = Camera.from_robot(robot, cam_name)
-        logging.info(f"found camera {cam_name}")
-
-        logging.info("displaying window")
-        while True:
-            # This is to stop this script just before the start of the next hour.
-            current_min = int(datetime.datetime.now().strftime("%M"))
-            if current_min == 59:
-                await close_robot(robot)
-                return
-
-            pil_img = await cam.get_image()
-            llist.append(datetime.datetime.now())
-
-            draw = ImageDraw.Draw(pil_img)
-            draw.rectangle((0, 0, 75, 25), outline='black')
-            fps = get_frames_per_sec(llist)
-            draw.text(xy=(18, 7.5), text=f'{fps} FPS', fill='white')
-
-            open_cv_image = cv2.cvtColor(np.array(pil_img), cv2.COLOR_RGB2BGR)
-            window_name = '2D'
-            cv2.namedWindow(window_name, cv2.WND_PROP_FULLSCREEN)
-            cv2.setWindowProperty(window_name, cv2.WND_PROP_FULLSCREEN, cv2.WINDOW_FULLSCREEN)
-            cv2.imshow(window_name, open_cv_image)
-            cv2.waitKey(1)
+        print("finished connecting to robot client")       
+
+        if arg == 1:
+            await image_stream(robot, llist1)
+        elif arg == 2:
+            await detection_stream(robot, llist2)
+        elif arg == 3:
+            task1 = asyncio.create_task(image_stream(robot, llist1))
+            task2 = asyncio.create_task(detection_stream(robot, llist2))
+            await task1
+            await task2
+        else:
+            print("unaccepted argument")
```

Comment:
[minor] Cool! The only thing I'm wondering is, do the screen sizes change based on the arguments too? It looks like `image_stream` and `detection_stream` are hard-coded to take up half the screen, which works well if they're concurrently displayed but not as well if only one is active.

Reply:
if one is active it is fullscreen

Reply:
Ah, I see!